library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInputOperator.java

```diff
@@ -0,0 +1,582 @@
+/**
+ * Copyright (C) 2015 DataTorrent, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datatorrent.lib.db.jdbc;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+
+import com.datatorrent.lib.util.FieldInfo;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * A concrete input operator that reads from a database through the JDBC API.<br/>
+ * This operator by default uses the {@link #fieldInfos} and the table name to construct the sql query.<br/>
+ *
+ * A user can provide there own query by setting the {@link #query} property which takes precedence.<br/>
+ *
+ * For eg. user can set the query property to a complex one : "select x1, x2 from t1, t2 where t1.x3 = t2.x3 ;"<br/>
+ *
+ * @displayName Jdbc Input Operator
+ * @category Input
+ * @tags database, sql, pojo, jdbc
+ * @since 2.1.0
+ */
+public class JdbcPOJOInputOperator extends AbstractJdbcInputOperator<Object> implements Operator.ActivationListener<Context.OperatorContext>
+{
+  private static int DEF_BATCH_SIZE = 100;
+
+  //table and query definition
+  protected String tableName;
```

Comment:
Why no \@NotNull annotation? Also if there are getters and setters the field should be private. This is because any modification to a non private field is potentially not backwards compatible (like adding an annotation or removing it).